rules/aip0136/http_uri_suffix.go

```diff
@@ -73,6 +76,24 @@ var uriSuffix = &lint.MethodRule{
 				}
 			}
 
+			// If the final component of the URI (before the verb) matches the final
+			// component of the RPC name, then assume this matches `nouns:verb`.
+			//
+			// Note that we do not do any pluralization here -- both the RPC and
+			// the URI must be plural already. If the RPC is singular, step 4 should
+			// apply.
+			plainURI := httpRule.GetPlainURI()
+			segs := strings.Split(strings.Split(plainURI, ":")[0], "/")
+			segment := segs[len(segs)-1]
+			if len(want) == 0 && strings.HasSuffix(strings.ToLower(n), segment) {
+				want = segment + ":" + strings.Split(strcase.SnakeCase(n), "_")[0]
+			}
+
+			// Nothing else applied; expect `:verbNoun`.
+			if len(want) == 0 {
+				want = ":" + strcase.LowerCamelCase(strcase.SnakeCase(m.GetName()))
```

Comment:
Why do we need to convert the name to snake case and then to lower camel case? And I think a helper function, like `toLowerCamelCase`, would help understanding and testing.